luacompiler/src/constants_map.rs

```diff
@@ -0,0 +1,115 @@
+use std::cell::RefCell;
+use std::collections::HashMap;
+
+/// Represents a structure which holds information about constants.
+/// For each type (int, float, string) there is a corresponding constant table in
+/// the bytecode.
+/// This structure creates a mapping between constants used in the Lua source code
+/// and their index in the constant table.
+///
+/// For instance, if '1.0' is used in Lua, then an entry from 1.0 to 0 is created in the
+/// <float_map> member.
+pub struct ConstantsMap {
```

Comment:
I'm not sure I understand why there's `int_count` and `int_maap` (isn't `int_count == int_map.len()`?); and whey we're using `RefCell` and `HashMap`. These are immutable constants so I assumed we'd simplify this to something like:

```rust
pub struct ConstantsMap { ints: Vec<i64>, floats: Vec<i64>, strings: Vec<String>}
```

Reply:
I changed this. I am using a map and not a vector because I map constants to indices, not the other way around.

Reply:
Ah, I see.